### ISSUE TITLE
Migrate to Java 11

### ISSUE THREAD

**Issue by ledsoft** [kbss-cvut/jopa#103]
Since Java 8 is past its lifetime and some of the libraries we depend on have or are going to require Java 11 (Jena 4.x, RDF4J 4.x, Openllet 2.6.5), it makes sense to migrate to Java 11 as well.

**Comment by ledsoft** [comment]
After considerations, the initial migration should be to Java 11. The main reason is that the AspectJ Maven plugin does not support Java 17 and later - https://github.com/mojohaus/aspectj-maven-plugin/issues/154. And while there is an alternative fork of this plugin, the MojoHaus plugin is definitely the most used one.

A subsequent ticket for migration to Java 17 will be created for version 2.0.0.

**Comment by kriegaex** [comment]
This is not true! Like I said in the Mojohaus thread, the AspectJ.dev versionj fully supports Java 17 since it was released. You can also use Java 18 or 19, whatever. In https://github.com/dev-aspectj/aspectj-maven-plugin/issues/113, you claimed it was not working without reacting to my comment requesting a minimal project as proof. Give me a reproducible problem, and I will help you. Simply stating here that is is unsupported, does not make it true.

**Comment by ledsoft** [comment]
@kriegaex I assume this is some misunderstanding. I did not create the [ticket](https://github.com/dev-aspectj/aspectj-maven-plugin/issues/113) you mention and I never participated in any [related discussion](https://github.com/mojohaus/aspectj-maven-plugin/issues/154).

Regarding the "alternative fork" of the AspectJ Maven plugin - I am well aware of the lack of development of the "official" AspectJ Maven plugin and the existence of your fork. I do, however, thank you for the idea of "a Maven plugin dictating my migration path". It gave me some food for thought. But since JOPA (this library) is not an end-user software, I try to make backward incompatible changes as infrequent and non-invasive as possible. I at least [updated the Wiki info](https://github.com/kbss-cvut/jopa/wiki/Maven-setup#java-17-and-later) on building with AspectJ and later JDKs.

**Comment by kriegaex** [comment]
Hm, that is a really strange coincidence then, two people
  * wanting the same thing,
  * making the same claim within (AspectJ Maven not working with Java 17),
  * commenting or pointing to the same Mojohaus issue
within a few days. I apologise if I thought that the other person is affiliated with your project.

An improvement idea for the [LTW chapter in your wiki](https://github.com/kbss-cvut/jopa/wiki/Maven-setup#load-time-weaving) is to mention that, according to the [AspectJ 1.9.7+ release notes](https://htmlpreview.github.io/?https://github.com/eclipse/org.aspectj/blob/master/docs/dist/doc/README-197.html), you need `--add-opens java.base/java.lang=ALL-UNNAMED` for Java 16+.

**Comment by kriegaex** [comment]
> I try to make backward incompatible changes as infrequent and non-invasive as possible

What is more backward incompatible in comparison when upgrading from Java 8 to 11 vs. 17? Or what is incompatible at all? Certainly not the change of a group ID and version number for a build plugin. How are your compatibility concerns related to this issue?

**Comment by ledsoft** [comment]
> Hm, that is a really strange coincidence then, two people
>
> - wanting the same thing,
> - making the same claim within (AspectJ Maven not working with Java 17),
> - commenting or pointing to the same Mojohaus issue
> - within a few days.

If you look at the history if this issue, it was not created within the last few days. I admit I revisited it recently because I received a GitHub notification regarding activity on the [Java 17 MojoHaus AspectJ Maven plugin issue](https://github.com/mojohaus/aspectj-maven-plugin/issues/154). Also, the referenced **MojoHaus AspectJ Maven** plugin (referenced in this issue) does not support Java 17, so I think at least my claim is correct.

Anyway, I did update all the Wiki documentation of JOPA to point to the `dev.aspectj` AspectJ Maven plugin, since it is recommended by the AspectJ developer guide. I also fixed my admittedly incorrect wording and replaced _official_ with _original_ w.r.t. the MojoHaus AspectJ Maven plugin.

> An improvement idea for the [LTW chapter in your wiki](https://github.com/kbss-cvut/jopa/wiki/Maven-setup#load-time-weaving) as a constructive take-away for you from this discussion is to mention that, according to the [AspectJ 1.9.7+ release notes](https://htmlpreview.github.io/?https://github.com/eclipse/org.aspectj/blob/master/docs/dist/doc/README-197.html), you need --add-opens java.base/java.lang=ALL-UNNAMED for Java 16+.

Thanks for the pointer, I updated the Wiki page accordingly.
